addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -36,15 +35,15 @@ func (r *VSphereCPIConfigReconciler) mapCPIConfigToDataValuesNonParavirtual( //
 	c := cpiConfig.Spec.VSphereCPI.NonParavirtualConfig
 	d.Mode = VsphereCPINonParavirtualMode
 
-	// get the vsphere cluster object
-	vsphereCluster, err := r.getVSphereCluster(ctx, cluster)
+	// get the vcenter values
+	vcenterValues, err := util.ParseVCenterValues(cluster)
 	if err != nil {
 		return nil, err
 	}
 
 	// derive the thumbprint, server from the vsphere cluster object
-	d.TLSThumbprint = vsphereCluster.Spec.Thumbprint
-	d.Server = vsphereCluster.Spec.Server
+	d.TLSThumbprint = vcenterValues.TlsThumbprint
```

Comment:
Question: Why not continue to rely on VsphereCluster object?

Reply:
It is just easier to retrieve all information from the variable vcenter. Otherwise we need to query VsphereCluster and VsphereMachineTemplate. If this is risky, we can go back to VsphereCluster.